app/src/main/java/com/onegini/mobile/exampleapp/view/activity/LoginActivity.java

```diff
@@ -192,6 +226,21 @@ private void setupUsersSpinner() {
     usersSpinner.setAdapter(spinnerArrayAdapter);
   }
 
+  private void loginUser(final UserProfile userProfile, final OneginiAuthenticator oneginiAuthenticator) {
+    OneginiSDK.getOneginiClient(this).getUserClient().authenticateUser(userProfile, oneginiAuthenticator, new OneginiAuthenticationHandler() {
```

Comment:
Duplicated code (getting user client, handler) in both login methods. Common parts could be extracted

Reply:
done